### ISSUE TITLE
When querying streams or stream IDs, ringserver only shows one stream

### ISSUE THREAD

**Issue by Guillaume-Savaton-ESEO** [EarthScope/ringserver#30]
I have compiled and set up ringserver 2020.075 for a home-made seismic interrogator.
The interrogator is based on an ARM 32-bit embedded computer that runs a customized Linux distribution based on the Yocto project.

The server is configured to scan a directory of miniSEED files generated with libmseed 2.19.
These files contains data from 4 channels.

When queried for the list of provided streams, using one of the methods below, the server always returns only one stream:

* `slinktool -Q server:18000`
* `slinktool -i STREAMS server:18000`
* `http://server:5000/streams`
* `http://server:5000/streamids`

However, `slinktook -p server:18000` reports data packets for all 4 channels.

Is it a configuration problem, or do you think my miniSEED files are ill-formed? I can provide some samples if needed.

Here is the content of the `ringserver.conf` file:

```
RingDirectory /media/data/ringserver
ListenPort 5000 HTTP
DataLinkPort 16000
SeedLinkPort 18000
ServerID "Loki"
TransferLogDirectory /media/data/log/
TransferLogRX 0
MSeedScan /media/data/record StateFile=/media/data/ringserver/ringserver.state Match=.*mseed$ InitCurrentState=y
HTTPHeader "Access-Control-Allow-Origin: *"
```

**Comment by chad-iris** [comment]
Hi @Guillaume-Savaton-ESEO,

Well, that is quite strange and, for the moment, I cannot imagine how that is possible.

Yes, please send me some examples of the data.  For more privacy, if you wish, you can send a pointer to a download to my email address on this page: https://www.iris.edu/hq/staff#trabant

**Comment by yensoon2** [comment]
Hi @chad-iris and @Guillaume-Savaton-ESEO,

I also can reproduce this issue, i.e. ringserver 2020.075 on my platform always returns only one stream.

The cause and proposed fix for this issue is described in pull request #29.

**Comment by chad-iris** [comment]
Thanks for connecting these issues @yensoon2.
In the next month or two I'll be working on ringserver again and will incorporate this fix in a release.

**Comment by chad-earthscope** [comment]
I belive this is resolved in v4.0.0
An internal hash function had numerical stability problems that only was problematic on certain platforms or certain levels of compiler optimization.

If you find this issue is still not fixed with v4.0.0 please reopen.
